site/static/diagrams/reusing-gameservers.puml

```diff
@@ -0,0 +1,64 @@
+@startuml
+participant Matchmaker
+participant Agones
+participant "Game Server\nProcess" as Binary
+participant SDK
+participant "GameServer\nResource" as GameServer
+box "Game Server Pod"
+  participant Binary
+  participant SDK
+end box
+
+== GameServer Start ==
+
+Agones -> GameServer: GameServer created through\na <i>Fleet</i> configuration
+activate GameServer
+GameServer -> Binary: Agones creates a Pod with the\nconfigured Game Server Container
+activate Binary
+activate SDK
+Binary -> SDK: SDK.Ready()
+note right
+  Call <i>Ready()</i> when the
+  Game Server can take player
+  connections and is able to
+  be allocated.
+end note
+GameServer <-- SDK: Update to <i>Ready</i> State
+
+== Matchmaker requesting GameServer ==
+loop <i>n</i> times
+Matchmaker -> Agones: Create: <i>GameServerAllocation</i>
+note left
+  Allocation atomically provides
+  a GameServer from a pool
+  of existing GameServers
+  (Usually a <i>Fleet</i>)
+end note
+"Agones" -> GameServer: Finds a Ready <i>GameServer</i>,\nsets it to <i>Allocated</i> State
+Matchmaker <-- Agones : <i>GameServerAllocation</i> is returned\nwith <i>GameServer</i> details\nincluding IP and port to connect to.
+
+== Players Finish Game ==
+
```

Comment:
I think it would make sense here to add a line from Binary -> Binary with a call like ResetGameServer() to show that this should happen _before_ the call to SDK.Ready().

Reply:
I like it. Done.